fleet_vehicle_inspection/models/fleet_vehicle_inspection.py

```diff
@@ -0,0 +1,196 @@
+        stored=True,
+        states=READONLY_STATES,
+    )
+
+    odometer_unit = fields.Selection(
+        [("kilometers", "Kilometers"), ("miles", "Miles")],
+        "Odometer Unit",
+        default="kilometers",
+        required=True,
+        states=READONLY_STATES,
+    )
+
+    date_inspected = fields.Datetime(
+        "Inspection Date",
+        required=True,
+        default=fields.Datetime.now,
+        help="Date when the vehicle has been inspected",
+        copy=False,
+        states=READONLY_STATES,
+    )
+
+    inspected_by = fields.Many2one(
+        "res.partner",
+        "Inspected By",
+        track_visibility="onchange",
+        states=READONLY_STATES,
+    )
+
+    direction = fields.Selection(
+        selection=[("in", "IN"), ("out", "OUT")], default="out", states=READONLY_STATES,
+    )
+
+    note = fields.Html("Notes", states=READONLY_STATES)
+
+    inspection_line_ids = fields.One2many(
+        "fleet.vehicle.inspection.line",
+        "inspection_id",
+        string="Inspection Lines",
+        copy=True,
+        auto_join=True,
+        states=READONLY_STATES,
+    )
+
+    result = fields.Selection(
+        [("todo", "Todo"), ("success", "Success"), ("failure", "Failure")],
+        "Inspection Result",
+        default="todo",
+        compute="_compute_inspection_result",
+        readonly=True,
+        copy=False,
+        store=True,
+    )
+
+    @api.depends("inspection_line_ids", "state")
+    def _compute_inspection_result(self):
+        for inspection in self:
+            if inspection.inspection_line_ids:
+                if inspection.inspection_line_ids.filtered(
+                    lambda x: x.result == "todo"
+                ):
+                    inspection.result = "todo"
+                    continue
+                elif inspection.inspection_line_ids.filtered(
+                    lambda x: x.result == "failure"
+                ):
+                    inspection.result = "failure"
+                else:
+                    inspection.result = "success"
+            else:
+                inspection.result = "todo"
+
+    @api.model
+    def create(self, vals):
+        if vals.get("name", "New") == "New":
+            if vals.get("direction") == "out":
+                vals["name"] = (
+                    self.env["ir.sequence"].next_by_code("fleet.vehicle.inspection.out")
+                    or "/"
+                )
+            else:
+                vals["name"] = (
+                    self.env["ir.sequence"].next_by_code("fleet.vehicle.inspection.in")
+                    or "/"
+                )
+        return super(FleetVehicleInspection, self).create(vals)
+
+    @api.multi
+    def button_cancel(self):
+        for inspection in self:
+            if inspection.state not in ["draft", "confirmed"]:
+                continue
+            inspection.write({"state": "cancel"})
+        return True
```

Comment:
Also nitpicking here.
You can do everything in 1 write

```suggestion
    def button_cancel(self):
        records = self.filtered(lambda rec: rec.state in ["draft", "confirmed"])
        return records.write({"state": "cancel"})
```

---

Although maybe it makes more sense to do something like this, instead.
It's usually more predictable behaviour

```suggestion
    def button_cancel(self):
        if any(rec.state not in ["draft", "confirmed"] for rec in self):
            raise ValidationError(_(
                "Only inspections in 'draft' or 'confirmed' states can be cancelled"
            ))
        return self.write({"state": "cancel"})
```